backoffice/backoffice/authors/api/views.py

```diff
@@ -249,7 +249,11 @@ def restart(self, request, pk=None):
                     workflow.id, workflow.workflow_type
                 )
                 error_msg = "No failed tasks found to restart. Skipping restart."
-            elif airflow_utils.find_failed_dag(workflow.id, workflow.workflow_type):
+            elif airflow_utils.find_failed_dag(
+                workflow.id, workflow.workflow_type
+            ) or not airflow_utils.find_executed_dags(
+                str(workflow.id), workflow.workflow_type
```

Comment:
Lets extract some variables in this `if - elif ` so as to make it nicer and more readable. For example 

```
restart_current_task = request.data.get("restart_current_task")

has_failed_dag = airflow_utils.find_failed_dag(
    workflow.id, workflow.workflow_type
)
has_no_executions = not airflow_utils.find_executed_dags(
    workflow.id, workflow.workflow_type
)
needs_initial_run_or_restart = has_failed_dag or has_no_executions
....
```

also no need to cast `workflow.id`  to str